### ISSUE TITLE
[FEA] Client script to export dask and dask-cuda configuration of  a dask-cuda cluster

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/dask-cuda#989]
## Request
We should have a script that a user can run on a client to export configuration and environment variables of  a dask-cuda/dask cluster . 

## Motivation:

Cluster configuration is often complex and requires the users to setup both in `dask-cuda` and `dask`   . It would really help if there was a way to get those exported easily into a file  to ensure that their setup is indeed correct. 

Common configuration errors seen are with the following configurations: 

##### Dask-Cuda Level:

- Transfer Protocol being userd (ucx/tcp) 
- Communication Hardware actually being used under the hood
-  Jit-Unspill 
- rmm-pool-size
- device-memory-limit

##### Dask Level 
 
- DASK_DISTRIBUTED__WORKERS__MEMORY__LIMIT
- DASK_DISTRIBUTED__COMM__TIMEOUTS__CONNECT

Potential UX:

```python
from dask_cuda import get_configration 
client = Client(.....)

get_configration(client, 'current_configuration.json')
```

Then the `current_configuration.json` can be easily shared with us to use. 

CC: @quasiben , @rlratzel , @randerzander , @jnke2016

**Comment by quasiben** [comment]
What do you think of the following:

## Cluster Setup 1

> cluster = LocalCUDACluster(rmm_pool_size='5GB', protocol='ucx')

```python
In [80]: pprint(list(client.run(lambda dask_worker: worker_config(dask_worker)).values())[0])
{'async_alloc': False,
 'distributed.comm.timeouts': {'connect': '30s', 'tcp': '30s'},
 'initial_pool_size': 5000000000,
 'jit-unspill': False,
 'libraries': [],
 'log_directory': None,
 'logging': False,
 'managed_memory': False,
 'maximum_pool_size': None,
 'memory_limit': 135263113216,
 'memory_pause_fraction': 0.8,
 'memory_spill_fraction': 0.7,
 'memory_target_fraction': 0.6,
 'protocol': 'ucx',
 'rmm_track_allocations': False,
 'ucx-transports': {'cma',
                    'cuda_copy',
                    'cuda_ipc',
                    'dc_mlx5',
                    'posix',
                    'rc_mlx5',
                    'rc_verbs',
                    'self',
                    'sysv',
                    'tcp',
                    'ud_mlx5',
                    'ud_verbs'}}
```

## Cluster Setup 2

> dask-cuda-worker ucx://10.33.227.165:8786 --rmm-pool-size 5GB --enable-jit-unspil

```python
In [52]: pprint(list(client.run(lambda dask_worker: worker_config(dask_worker)).values())[0])
{'async_alloc': False,
 'distributed.comm.timeouts': {'connect': '30s', 'tcp': '30s'},
 'initial_pool_size': 5000000000,
 'jit-unspill': True,
 'libraries': [],
 'log_directory': None,
 'logging': False,
 'managed_memory': False,
 'maximum_pool_size': None,
 'memory_limit': 135263113216,
 'memory_pause_fraction': 0.8,
 'memory_spill_fraction': 0.7,
 'memory_target_fraction': 0.6,
 'protocol': 'ucx',
 'rmm_track_allocations': False,
 'ucx-transports': {'cma',
                    'cuda_copy',
                    'cuda_ipc',
                    'dc_mlx5',
                    'posix',
                    'rc_mlx5',
                    'rc_verbs',
                    'self',
                    'sysv',
                    'tcp',
                    'ud_mlx5',
                    'ud_verbs'}}
```

## Spec Func
def worker_config(dask_worker):
    from distributed.comm import parse_address
    # assume homogenous cluster
    plugin_vals = dask_worker.plugins.values()
    ret = {}

    # device and host memory configuration
    for p in plugin_vals:
        for v in dir(p):
            # ignore hidden attrs as well as setup and CPUAffinity
            if v.startswith('_') or v == 'setup' or v == 'cores': pass
            else:
                ret[v] = getattr(p, v)
    for mem in ['memory_limit', 'memory_pause_fraction',
            'memory_spill_fraction', 'memory_target_fraction']:
        ret[mem] = getattr(dask_worker, mem)

    # jit unspilling set
    ret['jit-unspill'] = hasattr(dask_worker.data,
            'spill_on_demand_initialized')

    # using ucx ?
    scheme, loc = parse_address(dask_worker.scheduler.address)
    ret['protocol'] = scheme
    if scheme == 'ucx':
        import ucp
        ret['ucx-transports'] = ucp.get_active_transports()

    # comm timeouts
    ret['distributed.comm.timeouts'] = dask.config.get('distributed.comm.timeouts')

    return ret

**Comment by VibhuJawa** [comment]
Thanks for looking into this.  I think this looks good for us to use for now.

**Comment by VibhuJawa** [comment]
@quasiben , Can we add getting  `worker-ttl` and `work-stealing`  config too please .

**Comment by quasiben** [comment]
closed via #1006
